File: src/main/java/com/test/student/Student.java
package com.test.student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    //alt + insert可以產生建構子
    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest() {
        int max = (english > math)? english : math;
//        if (english > math) {
//            max = english;
//        } else {
//            max = math;
//        }

        return max;
    }

    public void print() {
        int average = getAverage();
        System.out.print(name + "\t" + english + "\t" + math
                + "\t" + getAverage() + ((average >= 60 ? "PASS" : "FAILED")) );
        /*if(getAverage() >= 60) {
            System.out.println("\tPASS");
        } else {
            System.out.println("\tFAILED");
        }*/

        char grading = 'F';
        switch (average / 10) {
            case 10 :
                grading = 'A';
                break;
            case 9 :
                grading = 'A';
                break;
            case 8 :
                grading = 'B';
                break;
            case 7 :
                grading = 'C';
                break;
            case 6 :
                grading = 'D';
                break;
            default:
                grading = 'F';
        }
        System.out.println(grading);
    }

    public int getAverage() {
        return  (english + math) / 2;
    }


}

File: src/main/kotlin/com/test/kotlin/Student.kt
package com.test.kotlin

import java.util.*

fun main() {
    userInput();
}

class Student(var name:String?, var english:Int, var math: Int) {
    fun print() {
/*        print(name + "\t" + english + "\t" + math + "\t" + getAverage()
                    + "\t" + getPassOrFailed());
        println("\t" + grading());*/
        print("$name\t$english\t$math\t${getAverage()}\t${getPassOrFailed()}\t${grading()}");
    }

    fun getAverage() :Int {
        return (english + math) / 2
    }

    fun nameCheck() {
        println(name?.length);
        if (name != null)
            println(name?.length);
    }

    fun highest() : Int {
        var max = if(english > math) {
            println("english")
            english;
        } else {
            println("math")
            math;
        }

        return max;
    }

    fun getPassOrFailed() = if(getAverage() >=60) "PASS" else "FAILED";

/*    fun grading() : Char {
        var grading : Char = when(getAverage()) {
            in 90..100 -> 'A';
            in 80..89 -> 'B';
            in 70..79 -> 'C';
            in 60..69 -> 'D';
            else -> 'F';
        }
        return grading
    }*/
    // kt function 判斷式可以直接當作右邊
    fun grading() = when(getAverage()) {
        in 90..100 -> 'A';
        in 80..89 -> 'B';
        in 70..79 -> 'C';
        in 60..69 -> 'D';
        else -> 'F';
    }
}

private fun userInput() {
    val scanner = Scanner(System.`in`);//兩個重音符號加關鍵詞即可使用，如此的in
    println("please enter student name:")
    val name = scanner.next()

    println("please enter student english:")
    val english = scanner.nextInt()

    println("please enter student math:")
    val math = scanner.nextInt()

    val student = Student(name, english, math);
    student.print();
/*    student.nameCheck();
    println("HighScore:${student.highest()}");*/
}